### ISSUE TITLE
Global configurable timeouts

### ISSUE THREAD

**Issue by jalaziz** [scalikejdbc/scalikejdbc#862]
Currently, it seems that the only way to set a query timeout is to set the query timeout on each query or to capture the executor and cancel the underlying statement after a timeout.

It would be great if a default query timeout could be configured globally.

This could use the `setQueryTimeout` option, but it may be possible to handle this at the execution layer to avoid issues with drivers not supporting `setQueryTimeout`. I have not dug too deep into the scalikejdbc source code, but from what I have see, it seems the former may be easier.

**Comment by jalaziz** [comment]
The issue with `socketTimeout` is that it's really a last resort. You end up killing the connection instead of just the query. Everything I've read suggests using the query timeout as first line of defense and `socketTimeout` as a last line of defense.

**Comment by seratch** [comment]
I see. Fair enough.

**Comment by seratch** [comment]
If you have the setting globally, having it per a connection pool would be straight-forward.  

I have one concern about the possible implementation. `ConnectionPoolSettings` already has `connectionTimeoutMillis` to set timeout duration in milliseconds while queryTimeout in JDBC is in seconds. I am not comfortable with the inconsistency, but it would be acceptable for many people if we name it as `queryTimeoutSeconds`.

**Comment by jalaziz** [comment]
Yeah, I agree, the inconsistency is annoying. `queryTimeoutSeconds` is reasonable though.

I can try to take a stab at this.

**Comment by nanajia984** [comment]
how is going with queryTimeoutSeconds? we need that feature as well

**Comment by seratch** [comment]
We haven't implemented it yet. I'm open to pull requests from the community!

By the way, I revisited this now and came to wonder if there is an equivalent setting in HikariCP.  https://github.com/brettwooldridge/HikariCP

The Commons DBCP has `defaultQueryTimeout`. http://commons.apache.org/proper/commons-dbcp/configuration.html

If HikariCP doesn't have such, there is no other way to ignore the queryTimeout value with it (we can output warnings tho).
